enhancements/hypershift/self-managed-azure.md

```diff
@@ -0,0 +1,754 @@
+
+Note: The token-minter sidecar is only injected for self-managed Azure (not
+ARO-HCP), per `token-minter-container.go:59-60`.
+
+#### CSI Driver Token Minter Configuration
+
+The Azure Disk and Azure File CSI drivers require special handling for workload
+identity authentication on self-managed HyperShift clusters
+([OCPBUGS-63698](https://issues.redhat.com/browse/OCPBUGS-63698)). CPO passes the
+`HYPERSHIFT_IMAGE` environment variable to the Cluster Storage Operator (CSO),
+which then passes it to the CSI driver operators. This enables the CSI operators
+to inject a token-minter sidecar container into the CSI driver controller pods.
+
+The token-minter flow works as follows:
+
+1. **CPO → CSO**: CPO sets `HYPERSHIFT_IMAGE` on the cluster-storage-operator
+   deployment, pointing to the token-minter image from the release payload.
+
+2. **CSO → CSI Operators**: CSO passes `HYPERSHIFT_IMAGE` to the Azure Disk and
+   Azure File CSI driver operators.
+
+3. **CSI Operators → Driver Controllers**: When `HYPERSHIFT_IMAGE` is set, the
+   CSI operators inject a token-minter sidecar into the CSI driver controller
+   pods with the following configuration:
+   ```yaml
+   containers:
+   - name: token-minter
+     image: <HYPERSHIFT_IMAGE>
+     command: ["/usr/bin/control-plane-operator", "token-minter"]
+     args:
+     - "--token-audience=openshift"
+     - "--service-account-namespace=openshift-cluster-csi-drivers"
+     - "--service-account-name=<csi-driver-controller-sa>"
+     - "--token-file=/var/run/secrets/openshift/serviceaccount/token"
+     - "--kubeconfig=/etc/kubernetes/kubeconfig"
+   ```
+
+4. **Token Minting**: The token-minter creates guest cluster service account
+   tokens and writes them to the shared volume at
+   `/var/run/secrets/openshift/serviceaccount/token`.
+
+5. **CSI Driver Authentication**: The CSI driver reads the token and uses Azure
+   Workload Identity Federation to authenticate to Azure APIs.
+
+This approach differs from ARO HCP, which uses Azure Secrets Store CSI Driver
+with SecretProviderClass resources to mount credentials from Azure Key Vault.
+
+### Workflow Description
+
+**Platform Engineer** is a human user responsible for setting up the management
+cluster and deploying hosted clusters.
+
+**Cluster Administrator** is a human user responsible for provisioning and
+maintaining Azure infrastructure.
+
+**HyperShift Operator** is the software component that manages hosted cluster
+lifecycles.
+
+1. The cluster administrator provisions the required Azure infrastructure:
+   - Creates a resource group for persistent shared resources
+   - Creates Virtual Network (VNet) with appropriate subnets
+   - Configures Network Security Groups (NSGs)
+   - Sets up Azure DNS zones
+   - Creates storage accounts for OIDC and image registry
+   - Creates managed identities for OpenShift components
+
+2. The platform engineer installs the OpenShift management cluster on Azure
```

Comment:
Would be nice to have a statement on what type of management clusters are we expecting to support. So far we've been testing with standalone Azure OpenShift without Workload Idenitity - is that the only expected configuration we'll be supporting?